source/MaterialXCore/Definition.h

```diff
@@ -100,6 +100,24 @@ class NodeDef : public InterfaceElement
         return getAttribute(NODE_ATTRIBUTE);
     }
 
+    /// Return the element's output type.
+    const string& getType() const override
```

Comment:
Overrides the getType() method for NodeDefs. Returns multi-output when there is more than one output on a NodeDef, otherwise returns the type of the single available output. If there are no outputs throws an exception that there are no outputs.